importers/link_16_importer.py

```diff
@@ -0,0 +1,217 @@
+        line_time += timedelta(hours=self.current_hour)
+        # Offset the parsed time relative to the file's timestamp
+        line_time += self.base_timestamp
+
+        time_token.record(self.name, "timestamp", line_time)
+
+        name_token.record(self.name, "vessel name", name_token.text)
+
+        # -- TODO - Confirm that this is correct for this data
+        platform = self.get_cached_platform(
+            data_store, platform_name=name_token.text, change_id=change_id
+        )
+        # TODO - confirm the actual sensor type
+        sensor_type = data_store.add_to_sensor_types("Location-Satellite", change_id=change_id).name
+        privacy = get_lowest_privacy(data_store)
+        sensor = platform.get_sensor(
+            data_store=data_store,
+            sensor_name="GPS",
+            sensor_type=sensor_type,
+            privacy=privacy,
+            change_id=change_id,
+        )
+        # -- END TODO
+        state = datafile.create_state(data_store, platform, sensor, line_time, self.short_name)
+
+        location = Location(errors=self.errors, error_type=self.error_type)
+        lat_success = location.set_latitude_decimal_degrees(lat_degrees_token.text)
+        lon_success = location.set_longitude_decimal_degrees(lon_degrees_token.text)
+        if lat_success and lon_success:
+            state.location = location
+            combine_tokens(lon_degrees_token, lat_degrees_token).record(
+                self.name, "location", state.location, "decimal degrees"
+            )
+
+        elevation_valid, elevation = convert_distance(
+            altitude_token.text, unit_registry.foot, line_number, self.errors, self.error_type
+        )
+
+        if elevation_valid:
+            state.elevation = elevation
+            altitude_token.record(self.name, "altitude", state.elevation)
+
+        heading_valid, heading = convert_absolute_angle(
+            heading_token.text, line_number, self.errors, self.error_type
+        )
+        if heading_valid:
+            state.heading = heading
+            heading_token.record(self.name, "heading", heading)
+
+        speed_valid, speed = convert_speed(
+            speed_token.text,
+            unit_registry.foot_per_second,
+            line_number,
+            self.errors,
+            self.error_type,
+        )
+        if speed_valid:
+            state.speed = speed
+            speed_token.record(self.name, "speed", speed)
+
+        datafile.flush_extracted_tokens()
+
+    @staticmethod
+    def extract_timestamp(filename):
+        """Extracts a Link-16 timestamp from an unmodified filename
+        :param filename: The Link-16 filename that contains a timestamp
+        :type filename: String
+        :return: The timestamp extracted from the filename
+        :rtype: String
+        """
+        base_filename, ext = os.path.splitext(os.path.basename(filename))
+        base_filename, ext = os.path.splitext(base_filename)
```

Comment:
This looks a bit unusual, but I'm guessing it's because there are two extensions on the end of the file? Like `blah.dat.txt`? If so, it's probably worth a comment explaining it.